src/Hydrator/NoopHydrator.php

```diff
@@ -24,6 +24,8 @@ final class NoopHydrator implements Hydrator
      * @param class-string $class
      *
      * @throws \LogicException
+     *
+     * @return never
```

Comment:
It seems unnecessary to me

Reply:
This helps Psalm and PHPStan recognize that not only does this function throw, but it specifically _always_ throws. So if you put any code after a call to this function, the static analyzers will know there's something wrong, because the code would never be executed. PHP 8.1 is actually adding a [native `never` return type](https://php.watch/rfcs/noreturn_type) for this reason.

Reply:
I understand, but this SDK is based on PHP 7.3 or greater, which means we should not use code that has been specifically added (or will be added) in future versions.

Reply:
Well, since it's just a comment annotation (rather than a native return type) it has no effect on the code's PHP 7.3 compatibility. But Psalm and PHPStan already support this annotation, so it will provide static analysis to everyone using those tools regardless of which PHP version they're using.